### PR TITLE
[SPARK-45335][SQL][DOCS] Correct the group of `ElementAt` and `TryElementAt`

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
Correct the group of `ElementAt` and `TryElementAt`, they both support array and map input, so should be in `collection functions`.

Existing category strategy seems like this: if a function support more than one types, it should be in `collection functions`:

- `Size` supports both array and map, it is in `collection functions`;
- `Reverse` supports both array and string, it is in `collection functions`;

So far, I didn't find other places with incorrect groups.

### Why are the changes needed?
for docs

### Does this PR introduce _any_ user-facing change?
yes, they will be in `collection functions` in SQL references

### How was this patch tested?
CI

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala [hunk 1]
```diff
@@ -2333,7 +2333,7 @@ case class Get(
        b
   """,
   since = "2.4.0",
-  group = "map_funcs")
+  group = "collection_funcs")
```
Comment (LuciferYang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala:2336]: https://github.com/apache/spark/blob/2aa06fcf1607bbad9e09649e587493032e739e35/sql/catalyst/src/main/java/org/apache/spark/sql/catalyst/expressions/ExpressionDescription.java#L110-L116

From the comments of `ExpressionDescription#group`, it seems we shouldn't directly use `collection_funcs`. Perhaps we should amend this comment. On the other hand, maybe the grouping of other functions needs correction, for instance, should `Concat` be categorized under `array_funcs`?

Comment (zhengruifeng) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala:2336]: I don't think it means `collection_funcs` is forbidden.

`Concat` works with multiple types, including string, binary and array, so it's not just an array function;

As I mentioned above, it looks like a `collection function` supports collection type (map/array/struct/etc) and some other types.

**Comment by zhengruifeng** [comment]
cc @cloud-fan @HyukjinKwon too

**Review by HyukjinKwon — APPROVED** [review]

**Review by cloud-fan — APPROVED** [review]

**Comment by zhengruifeng** [comment]
thanks all, merged to master

**Pull request closed without merging**